grafast/grafserv/src/servers/hono/v4/index.ts

```diff
@@ -0,0 +1,254 @@
+import type { Context as HonoContext, Hono, MiddlewareHandler } from "hono";
+import type { StatusCode } from "hono/utils/http-status";
+
+import {
+  convertHandlerResultToResult,
+  GrafservBase,
+  makeGraphQLWSConfig,
+  normalizeRequest,
+  processHeaders,
+} from "../../../index.js";
+import type {
+  EventStreamHeandlerResult,
+  GrafservBodyJSON,
+  GrafservConfig,
+  RequestDigest,
+  Result,
+} from "../../../interfaces.js";
+import { GRAPHQL_TRANSPORT_WS_PROTOCOL, makeServer } from "graphql-ws";
+import { UpgradeWebSocket } from "hono/ws";
+import { web } from "webpack";
+
+declare global {
+  namespace Grafast {
+    interface RequestContext {
+      Hono: {
+        honoContext: HonoContext;
```

Comment:
Is it necessary to name this `honoContext`? In the Koa adaptor we just call it `ctx` because that's how the Koa docs refers to it.

Reply:
Most of the hono doc uses just "c" but isn't it too obscure? ctx seems a good middleground indeed.

Reply:
I would use whatever hono docs readers would expect, so c is fine.